code/datums/components/deployable_item.dm

```diff
@@ -8,14 +8,17 @@
 	var/obj/deploy_type
 	///Any extra checks required when trying to deploy this item
 	var/datum/callback/deploy_check_callback
+	///Helps to determine if the item should be deployable in areas like the tad and alamo
+	var/restricted_deployment = FALSE
 
-/datum/component/deployable_item/Initialize(_deploy_type, _deploy_time, _undeploy_time, _deploy_check_callback)
+/datum/component/deployable_item/Initialize(_deploy_type, _deploy_time, _undeploy_time, _deploy_check_callback, _restricted_deployment)
```

Comment:
```suggestion
/datum/component/deployable_item/Initialize(_deploy_type, _deploy_time, _undeploy_time, _deploy_check_callback, _restricted_deployment = FALSE)
```

Because you've coded it to ALWAYS set restricted_deployment to this arg, its important to just explicitly set the default. The default value for the var itself on line 12 is redundant for this reason, and can be removed since it will never apply.